File: Java/src/com/leetcode/Easy/AddBinary.java
package com.leetcode.Easy;

// 67. Add Binary
public class AddBinary {

    public String addBinary(String a, String b) {
        StringBuilder result = new StringBuilder();
        int s = 0;
        int i = a.length() - 1, j = b.length() - 1;
        while (i >= 0 || j >= 0 || s == 1) {

            s += ((i >= 0) ? a.charAt(i) - '0' : 0);
            s += ((j >= 0) ? b.charAt(j) - '0' : 0);

            result.insert(0, (char) (s % 2 + '0'));

            s /= 2;

            i--;
            j--;
        }
        return result.toString();
    }
}

File: JavaScript/Easy/AddBinary.js
/**
 * 67. Add Binary
 * @param {string} a
 * @param {string} b
 * @return {string}
 */
var addBinary = function (a, b) {
  let result = "";
  let i = a.length - 1,
    j = b.length - 1,
    carry = 0;
  while (i >= 0 || j >= 0 || carry !== 0) {
    let x = i === -1 ? 0 : parseInt(a.charAt(i));
    let y = j === -1 ? 0 : parseInt(b.charAt(j));

    let sum = x + y + carry;
    if (sum === 3) {
      result = "1" + result;
      carry = 1;
    } else if (sum === 2) {
      result = "0" + result;
      carry = 1;
    } else {
      result = sum + result;
      carry = 0;
    }

    i = i === -1 ? -1 : --i;
    j = j === -1 ? -1 : --j;
  }

  return result;
};
